tiledb/sm/misc/utils.cc

```diff
@@ -276,6 +281,21 @@ Status get_fragment_name_version(const URI& uri, uint32_t* version) {
   return Status::Ok();
 }
 
+Status get_fragment_version(const std::string& name, uint32_t* version) {
+  uint32_t name_version;
+  RETURN_NOT_OK(get_fragment_name_version(name, &name_version));
+
+  if (name_version <= 2) {
+    *version = UINT32_MAX;
```

Comment:
This is a different function than the one used before. It is applicable only to later versions, where the name consists of 4 parts instead of 2 or 3, with the version being in the last `*_v` part. Not sure how to handle this, `UINT32_MAX` here stands for "invalid" or "not able to retrieve".